### PR TITLE
Add iOS functionality

### PR DESCRIPTION
**Pull request by david-alza**
- replaces contents in ios folder
- renames a few things
- adds and reworks example

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -17,10 +17,54 @@ const AlzaPaymentPass = NativeModules.AlzaPaymentPass
       }
     );
 
-export function canAddPaymentPass(): Promise<string> {
-  return AlzaPaymentPass.canAddPaymentPass();
+export function canAddPaymentPass(
+  uniqueCardReferenceID: string
+): Promise<string> {
+  return AlzaPaymentPass.canAddPaymentPass(uniqueCardReferenceID);
```
Comment (felipecsl) [inline src/index.tsx:23]: I think this will break android because it doesnt expect any arguments right now

Comment (felipecsl) [reply src/index.tsx:23]: maybe we should use separate `canAdd*` methods per platform?

Comment (david-alza) [reply src/index.tsx:23]: @felipecsl I think we actually want android to use the same reference too and then hook it into this here to see if the debit card is in the wallet? https://developers.google.com/pay/issuers/apis/push-provisioning/android/reading-wallet#listtokens

Comment (david-alza) [reply src/index.tsx:23]: finalizeAddPassToAppleWallet

Comment (david-alza) [reply src/index.tsx:23]: I went ahead and added the android code for this @felipecsl

File: android/src/main/java/com/alzapaymentpass/AlzaPaymentPassModule.kt [hunk 2]
```diff
@@ -20,7 +20,8 @@ class AlzaPaymentPassModule(reactContext: ReactApplicationContext) :
   }
 
   @ReactMethod
-  fun canAddPaymentPass(promise: Promise) {
+  fun canAddPaymentPass(uniqueCardReferenceID: String, promise: Promise) {
```
Comment (felipecsl) [inline android/src/main/java/com/alzapaymentpass/AlzaPaymentPassModule.kt:23]: @david-alza should we compare `uniqueCardReferenceID` to `token.getIssuerTokenId()` like in the docs you sent here? https://developers.google.com/pay/issuers/apis/push-provisioning/android/reading-wallet#listtokens

File: example/ios/Podfile.lock [hunk 3]
```diff
@@ -626,4 +626,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 4e9b3fb60f409ce0d6afe4fedcdb636e6fd38a9b
 
-COCOAPODS: 1.12.0
+COCOAPODS: 1.11.3
```
Comment (felipecsl) [inline example/ios/Podfile.lock:629]: hmm you might want to update your local cocoapods to avoid the revert to older version here

**Review by felipecsl — APPROVED** [review]

File: android/src/main/java/com/alzapaymentpass/AlzaPaymentPassModule.kt (hunk 2 above)
Comment (david-alza) [reply android/src/main/java/com/alzapaymentpass/AlzaPaymentPassModule.kt:21]: Yeah let's do this as a followup though cause we're gonna wanna get in the weeds of testing out how exactly this works in a real wallet.

File: example/ios/Podfile.lock (hunk 3 above)
Comment (david-alza) [reply example/ios/Podfile.lock]: reverted.

**Comment by felipecsl** [comment]
🚢

**Pull request merged**